### PR TITLE
Add machinery and docs for local generation of documentation

### PR DESCRIPTION
**Pull request by joshuagl**
Some changes to make generating documentation a bit nicer:
1) Add Gemfile to ease generating docs locally
2) Document the process for generating the documentation

### CODE REVIEW

File: docs/Gemfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+source 'https://rubygems.org'
```
[the remark this answers is not in the document]
Comment (MVrachev) [reply docs/Gemfile]: Can you explain more what is the process to generate docs locally?

**Comment by joshuagl** [comment]
I will split 2edf93d and c71324c into a separate PR.

I wasn't sure that docs/ was the right place, but we have other maintainer/developer centric docs there so it makes sense. I'll add something and update the PR.

File: docs/Gemfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshuagl) [reply docs/Gemfile]: Good catch, thanks

**Comment by joshuagl** [comment]
I've force-pushed to this PR to keep it focused on locally generating documentation. The other changes have been split into separate PR's #196 and #195.

**Comment by MVrachev** [comment]
It's a good idea to reference this new doc in the README or docs/index.md file or somewhere else.

Maybe in the manual deployment doc?

**Comment by joshuagl** [comment]
You're absolutely right, I was trying to avoid doing it until a) the relative links change landed and b) the docs relayout landed - but it would have been useful to mention that here.

I can do that now or (my preference) later, whichever you prefer.

**Comment by MVrachev** [comment]
> You're absolutely right, I was trying to avoid doing it until a) the relative links change landed and b) the docs relayout landed - but it would have been useful to mention that here.
> 
> I can do that now or (my preference) later, whichever you prefer.

Okay, let's do it later then. :))

**Comment by joshuagl** [comment]
Travis failed because Yarn timed out fetching a package, I've restarted the build.

**Pull request merged**